Update to Minecraft 1.21.3

JRoy opened the pull request:
~~All changes should be accounted for less items.json. Need to figure out experimental items filtering at runtime.~~

JRoy committed:
Update toolchain

JRoy committed:
Update to 1.12.3

JRoy committed:
1.12.3 Mob Changes

JRoy committed:
1.12.3 Item Changes

mdcfe commented on Essentials/src/main/java/com/earth2me/essentials/EssentialsPlayerListener.java:
```diff
@@ -609,7 +609,7 @@ public void onPlayerBucketEmpty(final PlayerBucketEmptyEvent event) {
     @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
     public void onPlayerCommandPreprocess(final PlayerCommandPreprocessEvent event) {
-        final String cmd = event.getMessage().toLowerCase(Locale.ENGLISH).split(" ")[0].replace("/", "").toLowerCase(Locale.ENGLISH);
+        final String cmd = event.getMessage().split(" ")[0].replace("/", "").toLowerCase(Locale.ENGLISH);
```
What does this fix?

JRoy replied:
nothing but why lower case the string two times

mdcfe commented on Essentials/src/main/java/com/earth2me/essentials/MobCompat.java:
```diff
@@ -299,35 +292,4 @@ private Villager.Profession asEnum() {
         }
     }
-
-    public enum BoatVariant {
```
Does this drop support for non-oak boats below 1.21.3?

JRoy replied:
did not test

JRoy committed:
Support boat variants on older versions

New versions will need to use the specific entity, as we cannot swap the entity type after it has been spawned

JRoy committed:
Fix compile errors from Sound no longer being an enum

JRoy committed:
Fix typo in build logic
